### PR TITLE
search contexts: Copy full query shortcut and replace copy icon

### LINKED ISSUE #18216: Replace "copy query" button icon with more representative icon
## Problem

The current icon in the "copy query" button was carried over from the experimental feature, and actually more accurately suggests "open in new window" than "copy."

CleanShot 2021-02-12 at 16 21 44@2x

## Solution

Replace icon in "copy query" button with the MDI `clipboard-outline` icon. Rendered icon (actual icon, not its container) should be 16x16.

CleanShot 2021-02-12 at 16 29 10@2x

### LINKED ISSUE #18318: Implement full query copy (including context)
Implement CMD+Shift+C shortcut to copy context with the query. (Ctrl+Shift+C on Windows/Linux)

### PR DESCRIPTION
**Pull request by novoselrok**
Fixes #18216 and fixes #18318

### CODE REVIEW

**Review by quinnkeast — APPROVED** [review]

File: client/web/src/search/input/toggles/CopyQueryButton.tsx [hunk 1]
```diff
@@ -1,42 +1,60 @@
...
+                type="button"
+                className={classNames('btn btn-icon icon-inline  btn-link-sm', props.className)}
+                data-tooltip={copied ? 'Copied!' : copyFullQueryTooltip}
+                onClick={nextClick}
+            >
+                <ClipboardOutlineIcon size={16} className="icon-inline" />
+            </button>
+            {props.keyboardShortcutForFullCopy.keybindings.map((keybinding, index) => (
+                <Shortcut key={index} {...keybinding} onMatch={copyFullQuery} allowDefault={false} ignoreInput={true} />
+            ))}
```
Comment (limitedmage) [inline client/web/src/search/input/toggles/CopyQueryButton.tsx:57]: Does anything need to be focused for this to work? Or just having the copy button visible will make this a global keyboard shortcut?

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/input/toggles/CopyQueryButton.tsx (hunk 1 above)
Comment (novoselrok) [reply client/web/src/search/input/toggles/CopyQueryButton.tsx:57]: Nothing needs to be focused, when the copy button is visible it should be registered as a global shortcut.

File: client/web/src/keyboardShortcuts/keyboardShortcuts.ts [hunk 2]
```diff
@@ -25,6 +25,15 @@ export const KEYBOARD_SHORTCUT_FOCUS_SEARCHBAR: KeyboardShortcut = {
     keybindings: [{ ordered: ['/'] }],
 }
 
+export const KEYBOARD_SHORTCUT_COPY_FULL_QUERY: KeyboardShortcut = {
+    id: 'copyFullQuery',
+    title: 'Copy full query',
+    keybindings: [
+        { held: ['Control', 'Shift'], ordered: ['c'] },
+        { held: ['Meta', 'Shift'], ordered: ['c'] },
```
[the remark this answers is not in the document]
Comment (novoselrok) [reply client/web/src/keyboardShortcuts/keyboardShortcuts.ts]: Yep, I agree, having extraneous keybindings feels a bit sloppy. I'll refactor this to only include the necessary keybindings for specific platform.

File: client/web/src/search/input/toggles/CopyQueryButton.tsx [hunk 3]
```diff
@@ -1,42 +1,60 @@
...
                     switchMapTo(merge(of(true), of(false).pipe(delay(2000)))),
                     tap(() => Tooltip.forceUpdate()),
                     startWith(false)
                 ),
-            [props.fullQuery]
+            [copyFullQuery]
         )
     )
 
+    const copyFullQueryTooltip = `Copy full query\n${window.navigator.platform.includes('Mac') ? '⌘' : 'Ctrl'}+⇧+C`
```
[the remark this answers is not in the document]
Comment (novoselrok) [reply client/web/src/search/input/toggles/CopyQueryButton.tsx]: You're right, I've extracted the platform detecting logic (simple boolean for now) to a util file and dependency injected it where needed (commit: https://github.com/sourcegraph/sourcegraph/pull/18317/commits/7cab7f4926e0a7e2efe52a7923fedd97e45aaebf). Take a look, it feels cleaner now.

**Review by limitedmage — APPROVED** [review]

**Pull request merged**